core/domain/services/admin/privacy/forms/PrivacySettingsFormHandler.php

```diff
@@ -0,0 +1,136 @@
+use EE_Full_HTML_Validation_Strategy;
+use EE_Registration_Config;
+use EE_Registry;
+use EE_Select_Reveal_Input;
+use EE_Text_Area_Input;
+use EEH_HTML;
+use EventEspresso\core\exceptions\InvalidFormSubmissionException;
+use EventEspresso\core\libraries\form_sections\form_handlers\FormHandler;
+use EventEspresso\core\services\loaders\LoaderFactory;
+
+/**
+ * Class PrivacySettingsFormHandler
+ *
+ * Class for displaying and processing the privacy settings form
+ *
+ * @package        Event Espresso
+ * @author         Mike Nelson
+ * @since          $VID:$
+ *
+ */
+class PrivacySettingsFormHandler extends FormHandler
+{
+
+    /**
+     * @var EE_Config
+     */
+    protected $config;
+
+
+    /**
+     * PrivacySettingsFormHandler constructor.
+     *
+     * @param EE_Registry $registry
+     * @param EE_Config   $config
+     */
+    public function __construct(EE_Registry $registry, EE_Config $config)
+    {
+        $this->config = $config;
+        parent::__construct(
+            esc_html__('Privacy Settings', 'event_espresso'),
+            esc_html__('Privacy Settings', 'event_espresso'),
+            'privacy-settings',
+            '',
+            FormHandler::DO_NOT_SETUP_FORM,
+            $registry
+        );
+    }
+
+
+    /**
+     * creates and returns the actual form
+     *
+     * @return EE_Form_Section_Proper
+     */
+    public function generate()
+    {
+        /**
+         * @var $reg_config EE_Registration_Config
+         */
+        $reg_config = $this->config->registration;
+        return new EE_Form_Section_Proper(
+            array(
+                'name'        => 'privacy_consent_settings',
+                'subsections' => array(
+                    'privacy_consent_form_hdr' => new EE_Form_Section_HTML(
+                        EEH_HTML::h2(esc_html__('Privacy Policy Consent Settings', 'event_espresso'))
+                    ),
+                    'enable'                   => new EE_Select_Reveal_Input(
+                        array(
+                            'enable-privacy-consent' => esc_html__('Enabled', 'event_espresso'),
+                            'disable'                => esc_html__('Disabled', 'event_espresso'),
+                        ),
+                        array(
+                            'default'         => $reg_config->isConsentCheckboxEnabled() ? 'enable-privacy-consent'
+                                : 'disable',
+                            'html_label_text' => esc_html__('Privacy Consent Checkbox', 'event_espresso'),
+                            'html_help_text'  => esc_html__(
+                                'When enabled, a checkbox appears in the registration form requiring users to consent to your site\'s privacy policy.',
+                                'event_espresso'
+                            ),
+                        )
+                    ),
+                    'enable-privacy-consent'   => new EE_Form_Section_Proper(
```

Comment:
why is this a whole new form section?
why not just add this as another input?

Reply:
currently, the select-reveal input works best when the section-to-be-revealed is a proper formsection, not just a form input. I tried setting it to an input, and it does successfully hide-and-reveal the requested input, but the input's label and help text etc doesn't get hidden. I'm ok with that restriction because we'll often want to hide groups of inputs anyways, and I'm not sure how much refactoring of the layout strategies would be required to change this.

Reply:
OHHHHH kk... didn't connect the dots that it was done because of the select-reveal input. makes total sense now thnx